aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -114,6 +115,33 @@
    */
   public static final String S3FILEIO_ACL = "s3fileio.acl";
 
+  /**
+   * If {@link org.apache.iceberg.aws.glue.GlueCatalog} should use external lock or not, default to false.
+   * If set to true, it will use DynamoDB to enforce locking during commits.
+   */
+  public static final String GLUE_CATALOG_LOCK_ENABLED = "gluecatalog.lock.enabled";
+  public static final boolean GLUE_CATALOG_LOCK_ENABLED_DEFAULT = false;
+
+  /**
+   * The DynamoDB table used for locking.
+   * One lock table is designed to be used for only one catalog.
+   * It is recommended to use a different table name for each Glue catalog.
+   * If the table does not exist, it will be created at runtime.
+   */
+  public static final String GLUE_CATALOG_LOCK_TABLE = "gluecatalog.lock.table";
+  public static final String GLUE_CATALOG_LOCK_TABLE_DEFAULT = "IcebergGlueCatalogLockTable";
+
+  /**
+   * After the given time in milliseconds, the process will give up the attempt to acquire a lock.
+   */
+  public static final String GLUE_CATALOG_LOCK_WAIT = "gluecatalog.lock.wait";
+  public static final long GLUE_CATALOG_LOCK_WAIT_DEFAULT = TimeUnit.MINUTES.toMillis(3);
+
+  /**
+   * After the given time in milliseconds, the lock is automatically expired.
+   */
+  public static final String GLUE_CATALOG_LOCK_EXPIRE = "gluecatalog.lock.expire";
+  public static final long GLUE_CATALOG_LOCK_EXPIRE_DEFAULT = TimeUnit.MINUTES.toMillis(2);
```

Comment:
Commits can take longer than this. What about using the max retry time for commits?

Reply:
Oh yes, I was planning to set it to 1200000ms which is `hive.zookeeper.session.timeout`, somehow typed it wrong...

Reply:
I was thinking `commit.retry.max-wait-ms` from table properties, which is the maximum amount of time before an operation fails and will stop retrying.

Reply:
Do you mean `commit.retry.total-timeout-ms`? `commit.retry.max-wait-ms` is only 60000ms which is 1 minute.